### DISCUSSION TITLE
I want to contribute the Yimmenu bu idk where to start.

### DISCUSSION THREAD

**Discussion by Skysea45** [YimMenu/YimMenu#3110] [Q&A] [answered]
First of all, I want to thanks to community for developing such a good solution. The biggest reason for me is that it's free and allows me to avoid dumb cheaters somewhat, and usually join sessions with a targeted player count range in regions with a better player profile.

Getting to the point, I want to contribute to this, but I don't know where to start. I've been using paid menus since 2016, not very often, and I have some experience with how cheats work and certain communication concepts in GTA Online. But I don't have proper knowledge about developing a script. I would be very happy if you could guide me on this.

I have a few questions in this regard.
1- Do we need to know/master 😳 on assembly language while reverse engineering to access low level elements in the game to develop a lua script?
2- Can good work be done using only .lua to develop protection systems where, for example, I think network access is important with the script?
3- What do you suggest I do to improve myself in this regard?

Thanks in advance for your help on this, both for me and for others in the community who want to contribute.

**Comment by rkwapisz** [comment, marked as answer]
1. For Lua? No, not at all. Check out the https://github.com/YimMenu-Lua repo and look at all the different scripts for examples.

2. Protections generally work by hooking into game functions that can be abused by other menus to generate corrupt/invalid messages that can crash clients. The YimMenu DLL is capable of directly patching GTA's memory whereas Lua scripts are not.

3. If you have working C++ knowledge you can always help identify areas in YimMenu's code where optimizations can be made to improve stability and performance. Alternatively, you can work through the https://github.com/YimMenu/Translations repo. One stretch goal I had was to ensure better consistency in the translation descriptions - proper title/heading capitalization, proper grammar and punctuation, etc.

**Reply by xesdoog** [reply]
Pretty sure Lua can [patch memory](https://github.com/YimMenu/YimMenu/blob/master/docs/lua/classes/lua_patch.md) through YimMenu.
